### ISSUE TITLE
Truly multi-view datasets

### ISSUE THREAD

**Issue by bkj** [Philip-Bachman/amdim-public#13]
In the paper, your method is motivated by learning from multiple views of the same dataset.  You've done experiments where you use data augmentation to create multiple synthetic views of the same training images, but have you done experiments / do you have thoughts on the best way to do experiments on datasets that are truly multiview?  

I'm thinking something like
 - adjacent frames in a video
 - two images taken from different angles

Do you think you'd still want to maximize mutual information between the global descriptor of view A and local descriptors of view B?  Or would it make more sense to just maximize MI between the global descriptors of the views?

Thanks!

**Comment by Philip-Bachman** [comment]
I think maximizing MI between higher and lower level features still makes sense in this case. When the InfoNCE matching/scores are computed in a relatively high-dimensional space (compared to the eg 128d/256d spaces used in most current contrastive rep learning code), rather complicated conditional distributions over the lower level features can be captured by the simple "dot products and softmaxes" approach. For more insight, see e.g. "Breaking the Softmax Bottleneck: a High Rank RNN Language Model" -- https://arxiv.org/abs/1711.03953
